ros2component/ros2component/api/__init__.py

```diff
@@ -0,0 +1,202 @@
+        (package_name, get_package_component_types(package_name=package_name))
+        for package_name in get_package_names_with_component_types()
+    ]
+
+
+ComponentInfo = namedtuple('Component', ('uid', 'name'))
+
+
+def get_container_components_info(*, node, remote_container_node_name):
+    list_nodes_client = node.create_client(
+        composition_interfaces.srv.ListNodes,
+        '{}/_container/list_nodes'.format(remote_container_node_name)
+    )
+    try:
+        list_nodes_client.wait_for_service()
+        future = list_nodes_client.call_async(
+            composition_interfaces.srv.ListNodes.Request()
+        )
+        rclpy.spin_until_future_complete(node, future)
+        response = future.result()
+        return [
+            ComponentInfo(uid, name) for uid, name in
+            zip(response.unique_ids, response.full_node_names)
+        ]
+    finally:
+        node.destroy_client(list_nodes_client)
+
+
+def load_component_into_container(
+    *,
+    node,
+    remote_container_node_name,
+    package_name,
+    plugin_name,
+    node_name=None,
+    namespace_name=None,
+    log_level=None,
+    remap_rules=None,
+    parameters=None,
+    extra_arguments=None
+):
+    load_node_client = node.create_client(
+        composition_interfaces.srv.LoadNode,
+        '{}/_container/load_node'.format(remote_container_node_name)
+    )
+    try:
+        load_node_client.wait_for_service()
+        request = composition_interfaces.srv.LoadNode.Request()
+        request.package_name = package_name
+        request.plugin_name = plugin_name
+        if node_name is not None:
+            request.node_name = node_name
+        if namespace_name is not None:
+            request.namespace_name = namespace_name
+        if log_level is not None:
+            request.log_level = log_level
+        if remap_rules is not None:
+            request.remap_rules = remap_rules
+        if parameters is not None:
+            for param in parameters:
+                name, _, value = param.partition(':=')
+                param_msg = rcl_interfaces.msg.Parameter()
+                param_msg.value = get_parameter_value(string_value=value)
+                param_msg.name = name
+                request.parameters.append(param_msg)
+        if extra_arguments is not None:
+            for arg in extra_arguments:
+                name, _, value = arg.partition(':=')
+                arg_msg = rcl_interfaces.msg.Parameter()
+                arg_msg.value = get_parameter_value(string_value=value)
+                arg_msg.name = name
+                request.extra_arguments.append(arg_msg)
+        future = load_node_client.call_async(request)
+        rclpy.spin_until_future_complete(node, future)
+        response = future.result()
+        if not response.success:
+            return 'Failed to load component: ' + response.error_message.capitalize()
+        print('{} {}'.format(response.unique_id, response.full_node_name))
+    finally:
```

Comment:
It's probably worth adding an `except` clause to catch and report any failures. Otherwise, this might fail quietly.

Reply:
My mistake, any exceptions should be visible if not caught.